blockchain/fullblocktests/generate.go

```diff
@@ -2538,11 +2747,76 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	g.AssertTipNumRevocations(2)
 	rejected(blockchain.ErrRevocationsMismatch)
 
+	// Create block that has a revocation with more payees than expected.
+	//   ... -> brt1(24)
+	//                  \-> brt3(25)
+	g.SetTip("brt1")
+	g.NextBlock("brt3", outs[25], ticketOuts[25], func(b *wire.MsgBlock) {
+		g.AssertBlockRevocationTx(b, 10)
+		b.STransactions[10].TxOut = append(b.STransactions[10].TxOut,
+			b.STransactions[10].TxOut[0])
+	})
+	g.AssertTipNumRevocations(1)
+	rejected(blockchain.ErrSSRtxPayeesMismatch)
+
+	// Create block that has a revocation paying more than the original
+	// amount to the committed address.
+	//   ... -> brt1(24)
+	//                  \-> brt4(25)
+	g.SetTip("brt1")
+	g.NextBlock("brt4", outs[25], ticketOuts[25], func(b *wire.MsgBlock) {
+		g.AssertBlockRevocationTx(b, 10)
+		b.STransactions[10].TxOut[0].Value++
+	})
+	g.AssertTipNumRevocations(1)
+	rejected(blockchain.ErrSSRtxPayees)
+
+	// Create block that has a revocation using a corrupted pay-to-address
+	// script.
+	//   ... -> brt1(24)
+	//                  \-> brt5(25)
+	g.SetTip("brt1")
+	g.NextBlock("brt5", outs[25], ticketOuts[25], func(b *wire.MsgBlock) {
+		g.AssertBlockRevocationTx(b, 10)
+		b.STransactions[10].TxOut[0].PkScript[8] ^= 0x55
+	})
+	g.AssertTipNumRevocations(1)
+	rejected(blockchain.ErrSSRtxPayees)
+
+	// Create block that has a revocation for a voted ticket.
+	//
+	//   ... -> brt1(24)
+	//                  \-> brt6(25)
+	g.SetTip("brt1")
+	g.NextBlock("brt6", outs[25], ticketOuts[25], func(b *wire.MsgBlock) {
+		voteTx := b.STransactions[1]
```

Comment:
This test is not doing what the comment suggests.  It needs to create a revocation for a ticket that has already voted, not reference the output of a vote.

What I would suggest is exporting the `createRevocationtx` from `chaingen` (and change the signature) in the same way `CreateVoteTx` was, with the exception that the block is not needed.  Then add `createRevocationTxFromTicket`, again, just as was done for `createVoteTxFromTicket` and update things accordingly.

Then the test you want is as follows:

```Go
		// Loop backwards to get the ticket transaction associated with the
		// first vote in the block.
		voteTx := b.STransactions[0]
		ticketTxIn := voteTx.TxIn[1]
		ticketBlk := g.BlockByHash(&b.Header.PrevBlock)
		for ticketBlk.Header.Height != ticketTxIn.BlockHeight {
			ticketBlk = g.BlockByHash(&ticketBlk.Header.PrevBlock)
		}
		ticketTx := ticketBlk.STransactions[ticketTxIn.BlockIndex]

		// Create new revocation for the same ticket.
		revocation := g.CreateRevocationTx(ticketTx, ticketTxIn.BlockHeight,
			ticketTxIn.BlockIndex)
		b.AddSTransaction(revocation)
		b.Header.Revocations++
```